src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterApplicationBuilderExtensions.cs

```diff
@@ -21,38 +21,133 @@
+        /// cref="IApplicationBuilder"/> for chaining calls.</returns>
+        public static IApplicationBuilder UseOpenTelemetryPrometheusScrapingEndpoint(this IApplicationBuilder app, string path)
+        {
+            Guard.ThrowIfNull(path);
+            return UseOpenTelemetryPrometheusScrapingEndpoint(app, meterProvider: null, predicate: null, path: path, configureBranchedPipeline: null);
+        }
+
+        /// <summary>
+        /// Adds OpenTelemetry Prometheus scraping endpoint middleware to an
+        /// <see cref="IApplicationBuilder"/> instance.
+        /// </summary>
+        /// <remarks>Note: A branched pipeline is created for the supplied
+        /// <paramref name="predicate"/>.</remarks>
+        /// <param name="app">The <see cref="IApplicationBuilder"/> to add
+        /// middleware to.</param>
+        /// <param name="predicate">Predicate for deciding if a given
+        /// <see cref="HttpContext"/> should be branched.</param>
+        /// <returns>A reference to the original <see
+        /// cref="IApplicationBuilder"/> for chaining calls.</returns>
+        public static IApplicationBuilder UseOpenTelemetryPrometheusScrapingEndpoint(this IApplicationBuilder app, Func<HttpContext, bool> predicate)
+        {
+            Guard.ThrowIfNull(predicate);
+            return UseOpenTelemetryPrometheusScrapingEndpoint(app, meterProvider: null, predicate: predicate, path: null, configureBranchedPipeline: null);
+        }
+
+        /// <summary>
+        /// Adds OpenTelemetry Prometheus scraping endpoint middleware to an
+        /// <see cref="IApplicationBuilder"/> instance.
+        /// </summary>
+        /// <remarks>Note: A branched pipeline is created based on the <paramref
+        /// name="predicate"/> or <paramref name="path"/>. If neither <paramref
+        /// name="predicate"/> nor <paramref name="path"/> are provided then
+        /// <see cref="PrometheusExporterOptions.ScrapeEndpointPath"/> is
+        /// used.</remarks>
         /// <param name="app">The <see cref="IApplicationBuilder"/> to add
         /// middleware to.</param>
         /// <param name="meterProvider">Optional <see cref="MeterProvider"/>
         /// containing a <see cref="PrometheusExporter"/> otherwise the primary
         /// SDK provider will be resolved using application services.</param>
-        /// <returns>A reference to the <see cref="IApplicationBuilder"/> instance after the operation has completed.</returns>
-        public static IApplicationBuilder UseOpenTelemetryPrometheusScrapingEndpoint(this IApplicationBuilder app, MeterProvider meterProvider = null)
+        /// <param name="predicate">Optional predicate for deciding if a given
+        /// <see cref="HttpContext"/> should be branched. If supplied <paramref
+        /// name="path"/> is ignored.</param>
+        /// <param name="path">Optional path to use for the branched pipeline.
+        /// Ignored if <paramref name="predicate"/> is supplied.</param>
```

Comment:
I think I'd kinda prefer separate overloads... that is, one that takes a predicate and another that take a path. I guess 

I imagine you were trying to keep the overload permutations to a minimum, but passing in either null or a value that is ignored feels funny.

Side question... I've pondered when if ever we should consider adopting nullable reference types.

Reply:
mm... we can explore this further and decide if we want to add more overloads, to make usage looks less awkward.

Reply:
> I imagine you were trying to keep the overload permutations to a minimum, but passing in either null or a value that is ignored feels funny.

For the overload with everything specified I tried to capture all the behaviors in the XML comments but ya, don't totally love this. Went through a few iterations of different numbers of extensions, different optional parameters, etc. Nothing felt totally 100% solid.

> Side question... I've pondered when if ever we should consider adopting nullable reference types.

We should yes! I tried it once in a branch, lots and lots of effort to pull it off 😭 You can turn on nullable analysis per file, so that part is nice for rolling it out over time. But what I couldn't figure out was how to tell public api analyzer how to respect the per-file mode. It seemed like an all or nothing thing where we would have to do it all in one shot per project. Painful that way!